lib/membrane_http_adaptive_stream/hls.ex

```diff
@@ -263,4 +260,26 @@ defmodule Membrane.HTTPAdaptiveStream.HLS do
   end
 
   defp serialize_ll_hls_tags(false, _target_partial_duration), do: ""
+
+  defp serialize_preload_hint_tag(true, track) do
+    # hls.js currently does not support EXT-X-PRELOAD_HINT tag
+    case Qex.pop_back(track.segments) do
+      {:empty, _segments} ->
+        ""
+
+      {{:value, last_segment}, _segments} ->
+        {name, bytes} =
+          case last_segment do
+            %Segment{type: :partial, name: name, parts: parts} ->
+              {name, Enum.reduce(parts, 0, fn part, acc -> acc + part.size end)}
+
+            %Segment{type: :full} ->
+              {track.segment_naming_fun.(track) <> track.segment_extension, 0}
+          end
+
+        "#EXT-X-PRELOAD-HINT:TYPE=PART,URI=\"#{name}\",BYTERANGE-START=#{bytes}"
+    end
+  end
```

Comment:
I would try something like that: 
```suggestion
    case Qex.last(track.segments) do
  :empty -> ""

  %Segment{type: :partial, name: name, parts: parts} = last_segment ->
    bytes = Enum.reduce(parts, 0, fn part, acc -> acc + part.size end)
    "#EXT-X-PRELOAD-HINT:TYPE=PART,URI=\"#{name}\",BYTERANGE-START=#{bytes}"

  %Segment{type: :full} = last_segment ->
    name = track.segment_naming_fun.(track) <> track.segment_extension
    "#EXT-X-PRELOAD-HINT:TYPE=PART,URI=\"#{name}\",BYTERANGE-START=0"
end
```